### PR TITLE
New watch face: Tally counter (tally_face)

### PR DESCRIPTION
**Pull request by logomancer**
This PR adds a new face for a six-digit [tally counter](https://www.amazon.com/s?k=tally+counter), based on the `counter_face` code by Shogo Okamoto. The controls are simple; Press `ALARM` to increment, long press `ALARM` to reset. There are distinct confirmation tones for when the counter is advanced and when the counter is reset, either manually or by rolling the counter over past 999999.

Let me know if there's anything else you need.

### CODE REVIEW

File: movement/watch_faces/complication/tally_face.c [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    tally_state_t *state = (tally_state_t *)context;
+
+    switch (event.event_type) {
+        case EVENT_MODE_BUTTON_UP:
+            movement_move_to_next_face();
+            break;
+        case EVENT_LIGHT_BUTTON_DOWN:
+            movement_illuminate_led();
+	        break;
+	    case EVENT_ALARM_BUTTON_UP:
```
Comment (tahnok) [inline movement/watch_faces/complication/tally_face.c:56]: little nit: mind fixing the indentation here?

**Review by tahnok — APPROVED** [review]
Looks good to me, although my only concern is the overlap with the counter face functionality might be confusing. I'd probably merge this as is because I can see different people wanting to have either more or less things counted, or to opt out of the functionality in the counter face where it beeps out the current count.

Bigger picture, I feel that lots of faces are going to grow settings to accommodate different needs, but I think that's a bit more complicated to implement from a UX perspective right now

**Comment by logomancer** [comment]
I just added a few more commits to fix the indentation issue as well as fix spacing and comments.

As for the potential for confusion, I'd argue that it was already present before I submitted this, based on my own experience; when picking out faces for my Sensor Watch firmware build, I saw `counter_face` and assumed it was a tally counter, not a lap counter for jogging. I was perplexed by the functionality of `counter_face` until I saw the description in the source header. Maybe there's an opportunity for better documentation all around.

**Review by joeycastillo — APPROVED** [review]
Looks good! Just made one minor change to silence a compiler warning. 

Thanks for making this!

**Pull request merged**